File: backend/app/schemas/admin_analytics.py
"""Admin health-analytics aggregation contracts."""

from __future__ import annotations

from pydantic import BaseModel, Field


class AnalyticsOverview(BaseModel):
    total_employees: int = 0
    covered_employees: int = 0
    coverage_rate: float = 0.0
    attention_employees: int = 0
    attention_rate: float = 0.0
    abnormal_exam_employees: int = 0
    plan_participants: int = 0
    plan_participation_rate: float = 0.0


class HealthDistributionItem(BaseModel):
    level: str
    label: str
    count: int
    percent: float


class RiskRankingItem(BaseModel):
    risk_type: str
    label: str
    count: int
    percent: float


class DepartmentStat(BaseModel):
    department: str
    total: int = 0
    coverage_rate: float = 0.0
    healthy_rate: float = 0.0
    attention_count: int = 0
    attention_rate: float = 0.0
    top_risk: str | None = None
    sample_too_small: bool = False


class TrendPoint(BaseModel):
    period: str
    label: str
    value: float = 0.0
    evaluated: int = 0


class HealthAnalyticsSummary(BaseModel):
    period: str
    department: str | None = None
    overview: AnalyticsOverview = Field(default_factory=AnalyticsOverview)
    health_distribution: list[HealthDistributionItem] = Field(default_factory=list)
    risk_ranking: list[RiskRankingItem] = Field(default_factory=list)
    department_stats: list[DepartmentStat] = Field(default_factory=list)
    risk_trend: list[TrendPoint] = Field(default_factory=list)
    # All departments the filter dropdown should show, independent of the
    # currently selected department, so the dropdown can switch freely.
    available_departments: list[str] = Field(default_factory=list)

File: frontend/src/api/adminAnalytics.ts
import { request } from './request'

export interface AnalyticsOverview {
  total_employees: number
  covered_employees: number
  coverage_rate: number
  attention_employees: number
  attention_rate: number
  abnormal_exam_employees: number
  plan_participants: number
  plan_participation_rate: number
}
export interface HealthDistributionItem { level: string; label: string; count: number; percent: number }
export interface RiskRankingItem { risk_type: string; label: string; count: number; percent: number }
export interface DepartmentStat { department: string; total: number; coverage_rate: number; healthy_rate: number; attention_count: number; attention_rate: number; top_risk: string | null; sample_too_small: boolean }
export interface TrendPoint { period: string; label: string; value: number; evaluated: number }
export interface HealthAnalyticsSummary {
  period: string
  department?: string | null
  overview: AnalyticsOverview
  health_distribution: HealthDistributionItem[]
  risk_ranking: RiskRankingItem[]
  department_stats: DepartmentStat[]
  risk_trend: TrendPoint[]
  available_departments: string[]
}

export const getHealthAnalyticsSummary = (period: string, department?: string | null) =>
  request.get<HealthAnalyticsSummary>('/admin/health-analytics/summary', { params: { period, department: department || undefined } }).then(r => r.data)
